### PR TITLE
Implemented meanZTest

### PR DESCRIPTION
**Pull request by ghost**
Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Implemented meanZTest

Detailed description / Documentation draft:
I implemented [meanZTest](https://mathcracker.com/z-test-for-two-means) aggregate function. 

```
Usage: 
meanZTest(population_variance_x, population_variance_y, confidence_level)(sample_data, sample_index)

Returns:
(z_statistics, p_value, confidence_interval_low, confidence_interval_high)
```

### CODE REVIEW

**Comment by nikitamikhaylov** [comment]
@achimbab May I ask you to write all formulas in Latex format inside the comments? And could you leave a link to some paper there?

**Comment by achimbab** [comment]
@nikitamikhaylov 
OK, I will do that ASAP.

**Comment by achimbab** [comment]
@nikitamikhaylov 

I wrote all formulas in Latex format inside the comments at #a8a20b6.

**Formulas and Descriptions for Mean z-test**
http://statkat.com/stat-tests/two-sample-z-test.php

**Online calculator for Mean z-test**
https://mathcracker.com/z-test-for-two-means

File: docs/en/sql-reference/aggregate-functions/reference/meanztest.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+Applies mean z-test to samples from two populations.
+
+**Syntax**
+
+``` sql
+meanZTest(population_variance_x, population_variance_y, confidence_level)(sample_data, sample_index)
+```
+
+Values of both samples are in the `sample_data` column. If `sample_index` equals to 0 then the value in that row belongs to the sample from the first population. Otherwise it belongs to the sample from the second population.
+The null hypothesis is that means of populations are equal. Normal distribution is assumed. Populations may have unequal variance and the variances are known.
```
Comment (nikitamikhaylov) [inline docs/en/sql-reference/aggregate-functions/reference/meanztest.md:17]: Is is really means that population are equal? I thought that null hypothesis is that only means of two distributions are equal.

Comment (nikitamikhaylov) [reply docs/en/sql-reference/aggregate-functions/reference/meanztest.md:17]: What is the alternative? Two sided or one sided? E.g. mean1 != mean2 or mean1 >(<) mean2?

**Comment by nikitamikhaylov** [comment]
Can we also add a python test with scipy? Please take a look at `01558_ttest_scipy.python`.

File: src/AggregateFunctions/AggregateFunctionMeanZTest.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        Float64 mean_y = getMeanY();
+
+        /// z = \frac{\bar{X_{1}} - \bar{X_{2}}}{\sqrt{\frac{\sigma_{1}^{2}}{n_{1}} + \frac{\sigma_{2}^{2}}{n_{2}}}}
+        Float64 zstat = (mean_x - mean_y) / getStandardError(pop_var_x, pop_var_y);
+        if (!std::isfinite(zstat))
+        {
+            return {std::numeric_limits<Float64>::quiet_NaN(), std::numeric_limits<Float64>::quiet_NaN()};
+        }
+
+        Float64 pvalue = 2.0 * boost::math::cdf(boost::math::normal(0.0, 1.0), -1.0 * std::abs(zstat));
```
Comment (nikitamikhaylov) [inline src/AggregateFunctions/AggregateFunctionMeanZTest.cpp:37]: Ok. https://www.omnicalculator.com/statistics/p-value#how-to-find-p-value-from-z-score

File: docs/en/sql-reference/aggregate-functions/reference/meanztest.md (hunk 1 above)
Comment (achimbab) [reply docs/en/sql-reference/aggregate-functions/reference/meanztest.md:17]: The alternative is Two sided.

Comment (achimbab) [reply docs/en/sql-reference/aggregate-functions/reference/meanztest.md:17]: I found that the null hypothesis for z-test is the below.
```
Null hypothesis:

  The two sample z test tests the following null hypothesis (H0):
    H0: μ1 = μ2
  Here μ1 is the population mean for group 1, and μ2 is the population mean for group 2.
```
Could you please explain why do you think like that?

Comment (nikitamikhaylov) [reply docs/en/sql-reference/aggregate-functions/reference/meanztest.md:17]: I somehow misread (I don't know how). I thought that it was written that the null hypothesis is that the two distributions are equal.. Sorry, my fault

**Comment by achimbab** [comment]
@nikitamikhaylov 
I added 02158_ztest_cmp.python.
But, I couldn't find the two_sample_mean_ztest in the scipy. So I implemented simple `twosample_mean_ztest()` that use unpooled variance in 02158_ztest_cmp.python and the test compares `twosample_mean_ztest()` and `meanZTest()`.

**Comment by achimbab** [comment]
https://mathcracker.com/z-test-for-two-means
It is online calculator for two sample mean z-test using unpooled variance.
The results of meanZTest and the calculator are the same.

File: docs/en/sql-reference/aggregate-functions/reference/meanztest.md (hunk 1 above)
Comment (achimbab) [reply docs/en/sql-reference/aggregate-functions/reference/meanztest.md:17]: That's OK. I dont mind. Thank you.

**Review by nikitamikhaylov — APPROVED** [review]

**Pull request merged**